opm/parser/eclipse/EclipseState/EclipseState.cpp

```diff
@@ -46,80 +47,55 @@ namespace Opm {
 
     namespace GridPropertyPostProcessor {
 
-        class DistributeTopLayer : public GridPropertyBasePostProcessor<double>
-        {
-        public:
-            DistributeTopLayer(const EclipseState& eclipseState) :
-                m_eclipseState( eclipseState )
-            { }
+        std::vector< double >& distTopLayer( std::vector<double>& values, const Deck& m_deck, const EclipseState& m_eclipseState ) {
```

Comment:
It's not *necessary*, but comes down to a mix of taste and goal. Personally I don't think passing reference for mutation like this is a good thing at all (I'll happily take a look into changing that properly i.e. GridPropertyFunction doesn't mutate but rather return a new vector).

Other than that, I prefer returning the reference because it allows for `return subfunction(values)` where `subfunction` is some static function in a source file. Ignoring the return value effectively makes it void anyway.